packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -470,7 +470,7 @@ function shouldRestoreFocus(scopeRef: ScopeRef) {
     scope = scope.parent;
   }
 
-  return true;
+  return scope?.scopeRef === scopeRef;
```

Comment:
what was the case that caused us to need this?
all I can think is that scope was null?

Reply:
answered above